lib/pom.rb

```diff
@@ -73,8 +73,8 @@ def initialize( name, version, default_spec = true )
   #            :id => 'gem-staging' )
 
   plugin( :clean,
-          :filesets => [ { :directory => '${basedir}/ruby/gems/shared/specifications/default',
-                           :includes => [ '*' ] },
+          :filesets => [ { :directory => '${basedir}/ruby/gems',
```

Comment:
It couldn't be left as it is, because bundled gems will be installed to `ruby/gems/shared/gems` and theirs specifications will be installed to to `ruby/gems/shared/specifications`.

Second problem with this approach is that all bundled gems need to be added [here](https://github.com/jruby/jruby/pull/4083/files#diff-a5692734fb1e233c63e88629e6f7bd12L290) and this is easy to forget to update this list.
And I didn't understand how to handle `SNAPSHOT` versions in this list, because `${jruby-readline.version}` is equal to `1.1.dev-SNAPSHOT`, but gem path should be `jruby-readline-1.1.dev-java`.

Also please see https://github.com/jruby/jruby/issues/4084 so more state makes build process more unpredictable.